src/components/Code/example.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { trim } from 'lodash';
+import styled from 'styled-components';
+
+import { Example, Info } from '../../utils/example';
+import { Input } from '..';
+import Code from '.';
+
+const Highlight = styled.span`
+  color: ${props => props.theme.colors.accent.blue};
+`;
+
+export default class CodeExample extends React.Component {
+  state = {
+    value: '<ACCESS_KEY>',
+  };
+
+  onCopy = () => {
+    console.log('Copied');
+  };
+
+  onChange = e => {
+    this.setState({
+      value: e.target.value,
+    });
+  };
+
+  render() {
+    return (
+      <div>
+        <Input placeholder="Dynamic content" onChange={this.onChange} />
+
+        <Info>Long single line command</Info>
+        <Example>
+          <Code onCopy={this.onCopy}>
+            kubectl create secret generic -n weave cloudwatch
+            --from-literal=access-key-id=xyz
+            --from-literal=secret-access-key=abc
+          </Code>
+        </Example>
+
+        <Info>Multiline file</Info>
+        <Example>
+          <Code isFile onCopy={this.onCopy}>
+            {trim(`
+{
+  "AccessKey": {
+    "AccessKeyId": "<YOUR_ACCESS_TOKEN>",
+    "SecretAccessKey": "<YOUR_SECRET_ACCESS_KEY>"
+  }
+}
+          `)}
+          </Code>
+        </Example>
+
+        <Info>Multiline single command</Info>
+        <Example>
+          <Code onCopy={this.onCopy}>
+            {trim(`
+kubectl create secret generic -n weave cloudwatch \\
+  --from-literal=access-key-id=xyz \\
+  --from-literal=secret-access-key=abc
+          `)}
+          </Code>
+        </Example>
+
+        <Info>Dynamic content in template string</Info>
+        <Example>
+          <Code isFile onCopy={this.onCopy}>
+            {() =>
```

Comment:
Is there a use case for passing a function? In a sense that if it wasn't a function, something would not work (or be much more complicated)?

Reply:
Allows for using template string formatting with dynamic values as opposed to having to deal with indentation in JSX

Reply:
Ok I see, was wondering about the particular use case in Service UI, but found `FluxSetupKubectlCommand` :)